llvm/lib/IR/AsmWriter.cpp

```diff
@@ -1444,6 +1460,10 @@ static void WriteConstantInternal(raw_ostream &Out, const Constant *CV,
         // Reparse stringized version!
         if (APFloat(APFloat::IEEEdouble(), StrVal).convertToDouble() == Val) {
           Out << StrVal;
+
+          if (CFP->getType()->isVectorTy())
+            Out << ")";
```

Comment:
The fact that this has to be repeated in a bunch of branches is not great -- might make sense to factor out the core APFloat printing out of this function?